Source/ASMultiplexImageNode.mm

```diff
@@ -493,19 +485,38 @@ - (UIImage *)_bestImmediatelyAvailableImageFromDataSource:(id *)imageIdentifierO
 
 #pragma mark -
 
-/**
- @note: This should be called without _downloadIdentifierLock held. We will lock
- super to read our interface state and it's best to avoid acquiring both locks.
- */
+- (void)_updatePriorityOnDownloaderIfNeededWithDefaultPriority:(ASImageDownloaderPriority)defaultPriority
+{
+  ASAssertUnlocked(_downloadIdentifierLock);
+  if (_downloaderFlags.downloaderImplementsSetPriority) {
+    id downloadIdentifier;
+    {
+      ASDN::MutexLocker l(_downloadIdentifierLock);
```

Comment:
@nguyenhuy is there a codepath where the downloadIdentifier could become invalid or set by a different thread first, then clobbered by this thread when resumed? Seems like an unlikely scenario, but it may need to be a locked transaction for correctness.

Reply:
You two are right. This is a change in behavior. At first it seemed to me that we should retain the lock as short as possible, but on a second thought, it's much safer to hold the lock for the entire transaction. Gonna update the diff.